cmd/root.go

```diff
@@ -69,12 +69,14 @@ func Execute() {
 
 func init() {
 	rootCmd.PersistentFlags().BoolP("debug", "d", false, "Enable or disable debug mode")
+	rootCmd.PersistentFlags().IntP("parallel", "p", -1, "Number of clones to fetch in parallel")
```

Comment:
Why is the default `-1` here in the CLI?

Reply:
Could we instead just set the default to 1 and have the code operate as `parallel 1` unless we change the flag to more parallel cloners?

Reply:
It's a bit of a story, really:

The original idea was for `parallel` to simply be an `int` describing the max number of coroutines, with `1` naturally being "no parallelism" and `< 1` being a magic value meaning "1 coroutine per CPU".  Of course, when I wrote this out in YAML, it meant having a config value `parallel: 0` which, to a human, looks like "disable parallelism".

So in a vain attempt to avoid confusion, I tried for "`parallel` is the max number of coroutines, or `0` to turn it off, or `-1` to use a reasonable default value".  It seemed like a good idea at the time.

Having written it all out, I think it might be time for an impromptu design session: I think simplicity might be key here, and turning it into a `bool` value would be the most effective.  `true` means run 1 coroutine per-CPU, with a max of 8 (this should give all the benefits, without having to worry about inadvertent DoS attacks from machines with a zillion cores on them) and `false` means just 1 coroutine.

Thoughts?

Reply:
I like your suggestion of:

> true means run 1 coroutine per-CPU, with a max of 8 (this should give all the benefits, without having to worry about inadvertent DoS attacks from machines with a zillion cores on them) and false means just 1 coroutine.

We could also add --max-parallel flag, where we move the `GOMAXPROCS` calculation into cobra, but it could be overriden for those that wish to do that.

Reply:
> Having written it all out, I think it might be time for an impromptu design session

Sweet... shall I come into the office and whiteboard?   I'll only bug Chet a little :p

Reply:
I'm inclined to take a page from the Cult of Agile™ and implement that later as a follow-on feature if literally anybody asks for it, ever.

Reply:
Sounds good to me